errors.go

```diff
@@ -33,3 +37,24 @@ func IsTemporaryError(err error) bool {
 	}
 	return false
 }
+
+type MessageError struct {
+	err     error
+	message amqp.Delivery
+}
+
+func NewMessageError(message amqp.Delivery, err error) *MessageError {
+	return &MessageError{err, message}
```

Comment:
```suggestion
	return &MessageError{err: err, message: message}
```
Minor one, but good for practice

Reply:
Fiiiiine